docs/courses/robotics/3.5-computer-vision.md

```diff
@@ -0,0 +1,393 @@
+    # Preprocess the image
+    processed = preprocess_image(image)
+    
+    # Find contours
+    contours, _ = cv2.findContours(processed, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+    
+    # Filter and analyze contours
+    cells = []
+    for contour in contours:
+        area = cv2.contourArea(contour)
+        if 50 < area < 500:  # Assume yeast cells are within this area range
+            cells.append(contour)
+    
+    return cells
+
+def analyze_growth(previous_count, current_count):
+    growth_rate = (current_count - previous_count) / previous_count * 100 if previous_count > 0 else 0
+    return growth_rate
+
+def microscope_scan_and_analyze():
+    print("Starting area scan...")
+    scan_images = microscope.scan([2000, 2000], [-2000, -2000])
+    
+    total_cells = 0
+    for i, img_data in enumerate(scan_images):
+        # Convert image data to OpenCV format
+        img = Image.open(io.BytesIO(img_data))
+        img_cv = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
+        
+        # Detect yeast cells
+        cells = detect_yeast_cells(img_cv)
+        total_cells += len(cells)
+        
+        # Draw detection results
+        img_with_cells = img_cv.copy()
+        cv2.drawContours(img_with_cells, cells, -1, (0, 255, 0), 2)
+        
+        # Display results
+        cv2.imshow(f'Scan {i+1}/{len(scan_images)}', img_with_cells)
+        cv2.waitKey(1000)  # Display for 1 second
+        
+        print(f"Scan image {i+1}/{len(scan_images)}: Detected {len(cells)} yeast cells")
+    
+    cv2.destroyAllWindows()
+    return total_cells
+
+def main():
+    try:
+        previous_cell_count = 0
+        while True:
+            # Move microscope to initial position and focus
+            microscope.move(0, 0)
+            microscope.focus()
+            
+            # Scan and analyze
+            current_cell_count = microscope_scan_and_analyze()
+            
+            # Calculate growth rate
+            growth_rate = analyze_growth(previous_cell_count, current_cell_count)
+            
+            print(f"Total detected yeast cells: {current_cell_count}")
+            print(f"Growth rate: {growth_rate:.2f}%")
+            
+            previous_cell_count = current_cell_count
+            
+            # Wait for a period before scanning again
+            time.sleep(3600)  # Wait for 1 hour
+    
+    except KeyboardInterrupt:
+        print("Monitoring stopped")
+    finally:
+        microscope.end_connection()
+
+if __name__ == "__main__":
+    main()
+```
+
+You can integrate the motorized microscope with your **OpenCV** detection pipeline, enabling real-time analysis and precise control.
+
+---
+
+## 🚀 Quiz
+
+::::{tab-set}
+:sync-group: category
+
+:::{tab-item} Sp/Su 2024
+:sync: sp2024
+
+[Quiz URL]
+:::
+
+::::
+
+---
+
+## 📄 Assignment
+
+Create a script that uses **OpenCV** and **AprilTags** to detect regions of interest in a sample and control a motorized microscope to move to those regions.
```

Comment:
I think this still needs to be updated, since AprilTags and the microscope are two separate topics